### PR TITLE
Validating PUT operation request/response schemas

### PR DESCRIPTION
**Pull request by dsgouda**
Addressing issue https://github.com/Azure/autorest/issues/1952

### CODE REVIEW

File: Samples/2a-arm-validation/shell/stderr.txt [hunk 1]
```diff
@@ -6,6 +6,8 @@ ERROR (ListByOperationsValidation/M1003/SDKViolation): Operation Usage_List must
         Path: https://raw.githubusercontent.com/Azure/azure-rest-api-specs/master/arm-storage/2015-06-15/swagger/storage.json#$.paths["/subscriptions/{subscriptionId}/providers/Microsoft.Storage/usages"].get.operationId
 ERROR (OperationsAPIImplementationValidation/M3023/RPCViolation): Operations API must be implemented for '/providers/Microsoft.Storage/operations'.
         Path: https://raw.githubusercontent.com/Azure/azure-rest-api-specs/master/arm-storage/2015-06-15/swagger/storage.json#$.paths
+ERROR (PutRequestResponseValidation/M2017/SDKViolation): "A PUT operation request body schema must be the same as its 200 response schema. Operation: StorageAccounts_Create"
```
Comment (olydis) [inline Samples/2a-arm-validation/shell/stderr.txt]: Just a thought, scrolling through these messages: Maybe it makes sense to write what the schemas involved are (something like `A PUT operation request body schema (found 'ABC') must be the same as its 200 response schema (found 'XYZ'). Operation: StorageAccounts_Create`).
That way we know what we're dealing with even without looking at the Swagger :-)

Comment (dsgouda) [inline Samples/2a-arm-validation/shell/stderr.txt]: Feedback much appreciated!

Comment (veronicagg) [inline Samples/2a-arm-validation/shell/stderr.txt]: what is this sample file for? I haven't seen it before.

Comment (olydis) [inline Samples/2a-arm-validation/shell/stderr.txt]: @veronicagg I added new (more real life) examples and modified CI a bit: stdout, stderr and exitcode of the examples is recorded as well - both for people to see what, say, "activating validation" or "supressing" *really* means and for us to detect unwanted changes in our API (AutoRest has *way* too little end-to-end tests) :-)

Comment (olydis) [inline Samples/2a-arm-validation/shell/stderr.txt]: so in this very case, it already makes us think "wait, does this new validation rule really apply to this example?"

**Comment by dsgouda** [comment]
@olydis @veronicagg Is this better?
As a side note I'm curious whether this will upset the usual structure most RPs use for PUT operations i.e., using a "ModelCreateParameters" model as request body schema and "Model" as their response model schema. Do we still want this as an error?

**Comment by olydis** [comment]
@dsgouda I share that concern :-D

**Comment by dsgouda** [comment]
@ravbhatnagar @amarzavery Definitely need you rinputs here!

File: src/autorest-core/test/directive.ts [hunk 2]
```diff
@@ -33,7 +33,7 @@ import { Message, Channel } from "../lib/message";
     // muted run
     await autoRest.ResetConfiguration();
     await autoRest.AddConfiguration({ "azure-arm": true });
-    await autoRest.AddConfiguration({ directive: { suppress: ["AvoidNestedProperties", "ModelTypeIncomplete", "DescriptionMissing"] } });
+    await autoRest.AddConfiguration({ directive: { suppress: ["AvoidNestedProperties", "ModelTypeIncomplete", "DescriptionMissing", "PutRequestResponseValidation"] } });
```
Comment (veronicagg) [inline src/autorest-core/test/directive.ts]: what is this "add configuration" directive about?

Comment (dsgouda) [inline src/autorest-core/test/directive.ts]: Adds a suppression so we can enforce the assertions for no warnings are found in the document is true.

**Pull request merged**